### PR TITLE
Google search view

### PR DESCRIPTION
**Pull request by JustAnOtherCoder13**
return result fro search view work.

Is it the good implementation?
If I follow this way, i have to add cursor adapter to  add response on user write.
No need to query with retrofit?

### CODE REVIEW

File: app/src/main/java/com/picone/go4lunch/presentation/ui/fragment/MapsFragment.java [hunk 1]
```diff
@@ -148,30 +157,30 @@ private void updateLocationUI() {
...
+                        .icon(BitmapDescriptorFactory.fromBitmap(getBitmapFromVectorDrawable(getContext(), R.drawable.ic_restaurant_with_no_user))));
             }
-            mMap.setOnMarkerClickListener(marker -> {
-                mRestaurantViewModel.initSelectedRestaurant(marker.getTitle());
-                Navigation.findNavController(requireView()).navigate(R.id.restaurantDetailFragment);
-                return false;
-            });
+        }
+        mMap.setOnMarkerClickListener(marker -> {
+            mRestaurantViewModel.initSelectedRestaurant(marker.getTitle());
```
Comment (fodeguirassy) [inline app/src/main/java/com/picone/go4lunch/presentation/ui/fragment/MapsFragment.java:174]: Navigation should be launched in liveData subscription

File: app/src/main/java/com/picone/go4lunch/presentation/ui/main/MainActivity.java [hunk 2]
```diff
@@ -101,6 +137,24 @@ private void initMenuButton() {
         });
     }
 
+    private SearchView.OnQueryTextListener getOnQueryTextListener() {
```
Comment (fodeguirassy) [inline app/src/main/java/com/picone/go4lunch/presentation/ui/main/MainActivity.java:207]: It seems that you don't need the token anymore

File: app/src/main/java/com/picone/go4lunch/presentation/viewModels/RestaurantViewModel.java [hunk 3]
```diff
@@ -116,6 +128,30 @@ public void getRestaurantFromMaps(Location mCurrentLocation) {
...
+        List<Restaurant> filteredRestaurant = new ArrayList<>();
+        getPredictionInteractor.getPredictions(query,MAPS_KEY,String.valueOf(location.getLatitude()).concat(",").concat(String.valueOf(location.getLongitude())))
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe(predictionResponse -> {
+                    //Log.i("TAG", "getPrediction: "+predictionResponse.getPredictions().get(0).getPlaceId());
+                    for (Prediction response : predictionResponse.getPredictions()){
+                        String placeId = response.getPlaceId();
+                        for (Restaurant restaurant:allRestaurantsMutableLiveData.getValue()){
+
```
Comment (fodeguirassy) [inline app/src/main/java/com/picone/go4lunch/presentation/viewModels/RestaurantViewModel.java]: Why do you need this ?

**Review by fodeguirassy — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/picone/go4lunch/presentation/viewModels/RestaurantViewModel.java (hunk 3 above)
Comment (JustAnOtherCoder13) [reply app/src/main/java/com/picone/go4lunch/presentation/viewModels/RestaurantViewModel.java]: to check wich restaurant is in response and assign it to allRestaurant in order to update ui

File: app/src/main/java/com/picone/go4lunch/presentation/ui/main/MainActivity.java (hunk 2 above)
Comment (JustAnOtherCoder13) [reply app/src/main/java/com/picone/go4lunch/presentation/ui/main/MainActivity.java:207]: Yes not needed to call response but highly recomended in API documentation.
delete for now.

File: app/src/main/java/com/picone/go4lunch/presentation/ui/fragment/MapsFragment.java (hunk 1 above)
Comment (JustAnOtherCoder13) [reply app/src/main/java/com/picone/go4lunch/presentation/ui/fragment/MapsFragment.java:174]: yes make a clean and review before passing all nav in live data subscription.

Comment (fodeguirassy) [reply app/src/main/java/com/picone/go4lunch/presentation/ui/fragment/MapsFragment.java:174]: I don't see the change here

File: app/src/main/java/com/picone/go4lunch/presentation/ui/main/MainActivity.java [hunk 4]
```diff
@@ -90,7 +107,14 @@ private void initMenuButton() {
             mBinding.drawerLayout.close();
             switch (item.getItemId()) {
                 case R.id.your_lunch_drawer_layout:
-                    mNavController.navigate(R.id.restaurantDetailFragment);
+                    mRestaurantViewModel.getCurrentUser.observe(this,user -> {
+                        if (user.getUserDailySchedule()!=null){
```
Comment (fodeguirassy) [inline app/src/main/java/com/picone/go4lunch/presentation/ui/main/MainActivity.java]: Same here. Nav should be launched when selectedRestaurant is set

**Review by fodeguirassy — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/picone/go4lunch/presentation/ui/main/MainActivity.java (hunk 4 above)
Comment (JustAnOtherCoder13) [reply app/src/main/java/com/picone/go4lunch/presentation/ui/main/MainActivity.java]: ok every navigation on live data subscription

File: app/src/main/java/com/picone/go4lunch/presentation/ui/fragment/MapsFragment.java (hunk 1 above)
Comment (JustAnOtherCoder13) [reply app/src/main/java/com/picone/go4lunch/presentation/ui/fragment/MapsFragment.java:174]: done for every navigation

File: app/src/main/java/com/picone/go4lunch/presentation/viewModels/RestaurantViewModel.java [hunk 5]
```diff
@@ -276,37 +276,44 @@ private void updateUserDailySchedule() {
...
-                }
+                if (dailyScheduleDate.compareTo(today) < 0) usersWithDailySchedulePassed.add(user);
+            }
+
+        }
+        return usersWithDailySchedulePassed;
+    }
+
+    @SuppressWarnings("ResultOfMethodCallIgnored")
+    @SuppressLint("CheckResult")
```
Comment (fodeguirassy) [inline app/src/main/java/com/picone/go4lunch/presentation/viewModels/RestaurantViewModel.java:300]: This is useless. As a developer you may not want to care about wether you should delete a DB. In general, a database table is not meant to be deleted.

**Review by fodeguirassy — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/picone/go4lunch/presentation/viewModels/RestaurantViewModel.java (hunk 5 above)
Comment (JustAnOtherCoder13) [reply app/src/main/java/com/picone/go4lunch/presentation/viewModels/RestaurantViewModel.java:300]: it is just to update interestedUsers and users daily shedule. Perhaps not the good name of the methode?

**Pull request merged**